sympy/solvers/solveset.py

```diff
@@ -559,14 +389,13 @@ def _solve_real_trig(f, symbol):
         return ConditionSet(symbol, Eq(f_original, 0), S.Reals)
 
 
-def _solve_as_poly(f, symbol, solveset_solver, invert_func):
+def _solve_as_poly(f, symbol, solveset_solver, invert_func, domain=S.Complexes):
```

Comment:
The `solveset_solver` argument is not being used anywhere in the function body. Also it not required now, as specify the solver unambiguously using the `domain` agrument. When I originally wrote this function the `domain` argument was not present and to call the different solver I had to use the different function `solveset_real` and `solveset_complex`.